pkg/multiTenancyPlugins/naming/nameScoping.go

```diff
@@ -139,25 +141,24 @@ func CheckContainerReferences(cluster cluster.Cluster, tenantId string, containe
 	volumesFrom := containerConfig.HostConfig.VolumesFrom
 
 	// create affinity array
-	affinityRef := make([]string, 0)
+	affinityLabelRefs := make([]string, 0)
+	affinityContainerRefs := make([]string, 0)
+	affinityLabelIndexs := make([]int, 0)
+	affinityContainerIndexs := make([]int, 0)
+
 	env := containerConfig.Env
-	var affinityIndex int
-	affinityLabelFirst := false
 	for i, e := range env {
 		if strings.HasPrefix(e, "affinity:") {
 			if strings.HasPrefix(e, "affinity:image==") {
 				break // ignore affinity for images
 			} else if strings.HasPrefix(e, "affinity:container==") {
 				containerRefIndex := strings.Index(e, "affinity:container==") + len("affinity:container==")
-				affinityRef = append(affinityRef, e[containerRefIndex:])
-				affinityIndex = i
-				break
+				affinityContainerRefs = append(affinityContainerRefs, e[containerRefIndex:])
+				affinityContainerIndexs = append(affinityContainerIndexs,i)
 			} else { // affinity:<label>:<value>
 				labelRefIndex := strings.Index(e, "affinity:") + len("affinity:")
-				affinityRef = append(affinityRef, e[labelRefIndex:])
-				affinityIndex = i
-				affinityLabelFirst = true
-				break
+				affinityLabelRefs = append(affinityLabelRefs, e[labelRefIndex:])
+				affinityLabelIndexs = append(affinityLabelIndexs,i)
 			}
```

Comment:
I'm not sure you actually need two array for the affinity , you just need to add a flag to the struct that will allow you to know whether it's label ref or container ref. Once you have this flag you can eliminate ALL double loops that you have all over the code as the only place you actually do something different about it is in the extraction of the ID